### ISSUE TITLE
[Recurrent Complex] Roofed Forest Dynamic Trees Incompatability

### ISSUE THREAD

**Issue by IdrisQe** [ferreusveritas/Dynamic-Trees#133]
Thank you for the lovely mod! This is easily becoming one of my favourite new aesthetic mods, I love how the trees still fit in Minecraft while being that much nicer. The inclusion of spreading forests, growth mechanics, built-in treechopper and more just makes it all the better!

That said, I've noticed a strange issue... And when searching for it I noticed this was actually mentioned in the Wiki though I'm not aware of the specifics of it. Any and all trees that generate in the Roofed Forest biome, and seemingly only the Roofed Forest biome, don't spawn as dynamic trees, regardless of tree type. This is a pretty painfully glaring sight when it's sitting right next to a grove of beautiful dynamic trees.

So, if it's not too much trouble, would it be feasible to add support to the Roofed Forest, or is there some backend Vanilla nonsense going on for that one specific biome that completely stops it from being possible? 
If that is the case- and I know this is probably well beyond the scope of this mod- could a Roofed Forest biome be created in a matching fashion to other biomes, and have it replace the Vanilla Roofed Forest in order to allow it to work?

**Comment by ghost** [comment]
Roofed forest should be working.  What version are you using?
Edit:  Also, can you send a screenshot of the problem?

**Comment by IdrisQe** [comment]
Apologies for the slow response time.
I'm using v0.7.8 with v1.2 BoP addon, v1.1.0 TC addon, Rustic v1.0.9 and Forge v14.23.4.2760.
In addition to Biomes O Plenty (v7.0.1.2399) I also have Climactic Biome Placement (v1.3.1) but I don't think that would affect it.

This effect has happened in every single Roofed Forest I have found. I've recently noticed it also happens in the Biomes O' Plenty Redwood Forest, although I imagine that's intended since the Redwoods aren't feasible to implement until multi-block trunks.

As seen in the picture, the Dark Oak outside the Roofed Forest generates as a Dynamic Tree, while the Dark Oaks, Birch, and every other tree inside the Roofed Forest is Vanilla-styled.
![2018-09-21_18 48 11](https://user-images.githubusercontent.com/40530630/45912068-a9896080-bdd0-11e8-8b9f-af6a8ae51f41.png)

**Comment by ghost** [comment]
Lemme guess.. recurrent complex?

**Comment by IdrisQe** [comment]
Oh. Uh... Yes, but I've disabled all of its nature generation (specifically because the trees it adds don't fit in with the Dynamic Trees), and these look like vanilla trees, not any of the monstrosities it spawns, so I didn't even consider that it would be the cause, and even then, absolutely zero Dynamic Trees spawn into the Roofed Forest biome even alongside these other trees, so that's... weird. Does RC do something to the Roofed Forest?

**Comment by ghost** [comment]
Do me a favor and create a new world with recurrent complex disabled and see if you still have the same problem.  Recurrent Complex is known to cause issues like this with Dynamic Trees.

**Comment by IdrisQe** [comment]
Yeah, I just checked and saw an error report of exactly that in RC's github. I've reported it to them as well, although the opened issue hasn't been touched in 21 days so ehh...
I'll try that really fast, and I'll edit this message when I have a result. Should just be 10 minutes or so at maximum.

Edit: I think the picture speaks for itself. Sorry for the trouble, I never would have guessed that a structure mod with nature turned off was altering a biome that drastically.
![2018-09-21_21 02 16](https://user-images.githubusercontent.com/40530630/45913132-d5611200-bde1-11e8-99c2-b890482f07e3.png)

Edit 2: On that note, I do have another suggestion. Since the Roofed Forest is meant to be a really dark and dense forest, would it be possible to increase the natural spawn-density of trees in one? Or would that cause too much lag or other issues?
